### PR TITLE
fix: Don't return resource if no actions in Refract adapter

### PR DESCRIPTION
**Pull request by pksunkara**

### CODE REVIEW

File: src/adapters/refract/transformResource.coffee [hunk 1]
```diff
@@ -37,18 +37,7 @@ module.exports = (resourceElement, location, options) ->
     attributes = _.dataStructures(resourceElement)
     attributes = if _.isEmpty(attributes) then undefined else attributes[0]
 
-    return [
```
Comment (kylef) [inline src/adapters/refract/transformResource.coffee:40]: The lines above (`attributes = _.dataStructures(resourceElement)`) can be removed? and also the comment above is no longer applicable.

Comment (pksunkara) [reply src/adapters/refract/transformResource.coffee:40]: Done.

**Comment by kylef** [comment]
I think overall it is better to handle this as a resource even when there are no actions because it could break other MSON Data Structures when a resource containing a data structure is removed at this stage.

Although, I understand this is consistent with the behaviour in API Blueprint AST adapter where this is the case that they are removed. @pksunkara What do you think about improving our comparison script to compare the two Application ASTs to be able to ignore this case instead of changing the behaviour in this adapter?

**Comment by pksunkara** [comment]
Unfortunately, there is no concept of resource in Apiary Blueprint AST, which means we can't do anything about this. Everything is an action. We already have a Trello card for the problem you are talking about and me and @Baggz discussed about this.

One more argument against this is `method` is always expected and since there's no action we can't fill out the method. Current refract adapter gives a `GET` method which is false since the user hasn't written anything.

As I said, I have given a lot of thought about this and I think this is the best way to move forward and then solve the resource with data structures problem separately.

**Comment by kylef** [comment]
In that case 👍. In future please mention these kind of discussions and concerns in PR description, it's great to have these documented somewhere.

**Pull request merged**